src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -725,10 +713,6 @@ private static bool IsUseGenitiveForm(ReadOnlySpan<char> format, int index, int
                             // This means that '%' is at the end of the format string or
                             // "%%" appears in the format string.
                             //
-                            if (resultBuilderIsPooled)
```

Comment:
Just to confirm my understanding, I assume we're not returning the VSB to the pool because we don't reasonably expect apps to ship with bad format strings?

Reply:
A combination of that, and we're starting off with 256 chars of stack space, so you'd not only need a bad format string but one that resulted in super, super long dates in order to require growing into the pool.